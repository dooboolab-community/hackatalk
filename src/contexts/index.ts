import React from 'react';

export const AppContext = React.createContext<any>(null);
