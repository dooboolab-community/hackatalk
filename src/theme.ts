/* eslint-disable */
export const colors = {
  light: '#ffffff',
  dark: '#151A25',
  backgroundDark: '#151E22',
  deepDark: '#000000',
  blue_80: '#2F38C7',
  blue_70: '#1E6EFA',
  blue_60: '#217DFE',
  paleBlue_10: '#8bc9ff',
  paleBlue_05: '#e2f2ff',
  green_30: '#00D4AB',
  green_10: '#A1EEDB',
  gray_80: '#676c7a',
  gray_70: '#545966',
  gray_60: '#676c7a',
  gray_50: '#8f94a3',
  gray_40: '#AFB4C3',
  gray_30: '#d3d8e8',
  gray_05: '#f3f9ff',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: colors.light,
  backgroundDark: colors.paleBlue_05,
  itemBackground: colors.light,
  btnPrimary: colors.blue_70,
  btnPrimaryFont: colors.light,
  btnPrimaryLight: colors.light,
  btnPrimaryLightFont: colors.blue_70,
  textDisabled: colors.gray_50,
  btnDisabled: colors.gray_30,
  fontColor: colors.dark,
  fontSubColor: colors.blue_70,
  labelColor: colors.dark,
  tintColor: colors.green_30,
  lineColor: colors.gray_30,
  indicatorColor: colors.blue_70,
  inactiveColor: colors.gray_60,
  primary: colors.blue_70,
  primaryLight: colors.paleBlue_10,
  headerFont: colors.light,
  searchBackground: colors.gray_05,
  messageBackground: colors.light,
  status: colors.green_10,
  placeholder: colors.gray_60,
  focused: colors.blue_70,
  placeholderFocused: colors.blue_70,
  socialBtn: colors.paleBlue_05,
  link: colors.blue_70,
  myMessageBackground: colors.paleBlue_10,
  myMessageText: colors.dark,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.light,
  modalBtnPrimaryFont: colors.blue_80,
  modalBtnFont: colors.gray_60,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.light,
  facebookText: colors.light,
  facebookBackground: colors.facebook,
  googleIcon: colors.light,
  googleText: colors.light,
  googleBackground: colors.google,
  errorTitle: colors.gray_60,
  errorBody: colors.gray_50,
  fab: colors.blue_70,
};
export type Theme = typeof light;
export const dark = {
  background: colors.deepDark,
  backgroundDark: colors.deepDark,
  itemBackground: colors.dark,
  btnPrimary: colors.paleBlue_10,
  btnPrimaryFont: colors.deepDark,
  btnPrimaryLight: colors.dark,
  btnPrimaryLightFont: colors.paleBlue_10,
  textDisabled: colors.gray_80,
  btnDisabled: colors.gray_40,
  fontColor: colors.light,
  fontSubColor: colors.paleBlue_10,
  labelColor: colors.light,
  tintColor: colors.green_10,
  lineColor: colors.light,
  indicatorColor: colors.paleBlue_10,
  inactiveColor: colors.gray_50,
  primary: colors.deepDark,
  primaryLight: colors.dark,
  headerFont: colors.light,
  searchBackground: colors.gray_80,
  messageBackground: colors.dark,
  status: colors.green_30,
  placeholder: colors.gray_50,
  focused: '#ffffff',
  placeholderFocused: colors.paleBlue_10,
  socialBtn: colors.gray_05,
  link: colors.paleBlue_05,
  myMessageBackground: colors.deepDark,
  myMessageText: colors.light,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.gray_80,
  modalBtnPrimaryFont: colors.paleBlue_10,
  modalBtnFont: colors.gray_40,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.facebook,
  facebookText: colors.dark,
  facebookBackground: colors.light,
  googleIcon: colors.google,
  googleText: colors.dark,
  googleBackground: colors.light,
  errorTitle: colors.paleBlue_10,
  errorBody: colors.gray_40,
  fab: colors.paleBlue_10,
};
