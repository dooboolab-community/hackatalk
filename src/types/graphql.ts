export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  Date: any;
  DateTime: any;
  Upload: any;
};

export type AuthPayload = {
  __typename?: 'AuthPayload';
  token: Scalars['String'];
  user: User;
};

export enum AuthType {
  Email = 'email',
  Facebook = 'facebook',
  Google = 'google',
  Apple = 'apple'
}

export type Channel = {
  __typename?: 'Channel';
  id: Scalars['ID'];
  type?: Maybe<ChannelType>;
  name?: Maybe<Scalars['String']>;
  messages?: Maybe<Array<Maybe<Message>>>;
  memberships?: Maybe<Array<Maybe<Membership>>>;
  myMembership?: Maybe<Membership>;
  createdAt: Scalars['DateTime'];
  updatedAt: Scalars['DateTime'];
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type ChannelInput = {
  type?: Maybe<ChannelType>;
  name?: Maybe<Scalars['String']>;
  friendIds: Array<Scalars['String']>;
};

export enum ChannelType {
  Private = 'PRIVATE',
  Public = 'PUBLIC'
}



export type File = {
  __typename?: 'File';
  filename: Scalars['String'];
  mimetype: Scalars['String'];
  encoding: Scalars['String'];
};

export type Friend = {
  __typename?: 'Friend';
  id: Scalars['ID'];
  user?: Maybe<User>;
  friend?: Maybe<User>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type FriendPayload = {
  __typename?: 'FriendPayload';
  user: User;
  added?: Maybe<Scalars['Boolean']>;
  deleted?: Maybe<Scalars['Int']>;
};

export type FriendSub = {
  __typename?: 'FriendSub';
  user?: Maybe<User>;
  action?: Maybe<FriendSubAction>;
};

export enum FriendSubAction {
  Added = 'ADDED',
  Updated = 'UPDATED',
  Deleted = 'DELETED'
}

export type Gallery = {
  __typename?: 'Gallery';
  id: Scalars['ID'];
  photoURL: Scalars['String'];
  createdAt: Scalars['DateTime'];
  updatedAt: Scalars['DateTime'];
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export enum Gender {
  Male = 'MALE',
  Female = 'FEMALE'
}

export type Membership = {
  __typename?: 'Membership';
  id: Scalars['ID'];
  channel?: Maybe<Channel>;
  user?: Maybe<User>;
  type?: Maybe<MemberType>;
  userAlert?: Maybe<Scalars['Boolean']>;
  userMode?: Maybe<UserModeType>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export enum MemberType {
  Owner = 'OWNER',
  Member = 'MEMBER'
}

export type Message = {
  __typename?: 'Message';
  id: Scalars['String'];
  channel?: Maybe<Channel>;
  sender?: Maybe<User>;
  type?: Maybe<Scalars['String']>;
  text?: Maybe<Scalars['String']>;
  picture?: Maybe<Array<Maybe<Photo>>>;
  filePath?: Maybe<Scalars['String']>;
  replies?: Maybe<Array<Maybe<Reply>>>;
  reactions?: Maybe<Array<Maybe<Reaction>>>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type MessagePayload = {
  __typename?: 'MessagePayload';
  channelId: Scalars['String'];
  message?: Maybe<Message>;
};

export type Mutation = {
  __typename?: 'Mutation';
  signInEmail: AuthPayload;
  signInWithSocialAccount: AuthPayload;
  signInWithFacebook: AuthPayload;
  signUp: AuthPayload;
  findPassword?: Maybe<Scalars['Boolean']>;
  sendVerification?: Maybe<Scalars['Boolean']>;
  addNotificationToken?: Maybe<Notification>;
  removeNotificationToken?: Maybe<Scalars['Int']>;
  updateProfile?: Maybe<User>;
  addFriend?: Maybe<FriendPayload>;
  deleteFriend?: Maybe<FriendPayload>;
  /** `friendIds` in Channel should exclude userid. */
  createChannel?: Maybe<Channel>;
  updateChannel?: Maybe<Scalars['Int']>;
  deleteChannel?: Maybe<Scalars['Int']>;
  /**
   * Create message and return channelId when meessage has successfully sent.
   * Do not pass current userId inside `users`.
   */
  createMessage?: Maybe<MessagePayload>;
  setOnlineStatus?: Maybe<Scalars['Int']>;
  changeEmailPassword?: Maybe<Scalars['Boolean']>;
  createGallery?: Maybe<Gallery>;
  updateGallery?: Maybe<Scalars['Int']>;
  deleteGallery?: Maybe<Scalars['Int']>;
  singleUpload: Scalars['String'];
  createReaction?: Maybe<Reaction>;
  deleteReaction?: Maybe<Scalars['Int']>;
};


export type MutationSignInEmailArgs = {
  email: Scalars['String'];
  password: Scalars['String'];
};


export type MutationSignInWithSocialAccountArgs = {
  socialUser: SocialUserInput;
};


export type MutationSignInWithFacebookArgs = {
  accessToken: Scalars['String'];
};


export type MutationSignUpArgs = {
  user: UserInput;
};


export type MutationFindPasswordArgs = {
  email: Scalars['String'];
};


export type MutationSendVerificationArgs = {
  email: Scalars['String'];
};


export type MutationAddNotificationTokenArgs = {
  notification: NotificationCreateInput;
};


export type MutationRemoveNotificationTokenArgs = {
  token: Scalars['String'];
};


export type MutationUpdateProfileArgs = {
  user: UserProfileInput;
};


export type MutationAddFriendArgs = {
  friendId: Scalars['ID'];
};


export type MutationDeleteFriendArgs = {
  friendId: Scalars['ID'];
};


export type MutationCreateChannelArgs = {
  channel?: Maybe<ChannelInput>;
};


export type MutationUpdateChannelArgs = {
  channel?: Maybe<ChannelInput>;
};


export type MutationDeleteChannelArgs = {
  channelId: Scalars['ID'];
};


export type MutationCreateMessageArgs = {
  message: Scalars['String'];
  channelId: Scalars['String'];
};


export type MutationSetOnlineStatusArgs = {
  isOnline?: Maybe<Scalars['Boolean']>;
};


export type MutationChangeEmailPasswordArgs = {
  password: Scalars['String'];
  newPassword: Scalars['String'];
};


export type MutationCreateGalleryArgs = {
  photoURL: Scalars['String'];
};


export type MutationUpdateGalleryArgs = {
  galleryId: Scalars['ID'];
  photoURL: Scalars['String'];
};


export type MutationDeleteGalleryArgs = {
  galleryId: Scalars['ID'];
};


export type MutationSingleUploadArgs = {
  file: Scalars['Upload'];
  dir?: Maybe<Scalars['String']>;
};


export type MutationCreateReactionArgs = {
  messageId: Scalars['ID'];
  type: Scalars['String'];
};


export type MutationDeleteReactionArgs = {
  reactionId: Scalars['ID'];
};

export type Notification = {
  __typename?: 'Notification';
  id: Scalars['ID'];
  token?: Maybe<Scalars['String']>;
  device?: Maybe<Scalars['String']>;
  os?: Maybe<Scalars['String']>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
};

export type NotificationCreateInput = {
  token: Scalars['String'];
  device?: Maybe<Scalars['String']>;
  os?: Maybe<Scalars['String']>;
};

/** Information about pagination in a connection. */
export type PageInfo = {
  __typename?: 'PageInfo';
  startCursor?: Maybe<Scalars['String']>;
  endCursor?: Maybe<Scalars['String']>;
  hasNextPage?: Maybe<Scalars['Boolean']>;
  hasPreviousPage?: Maybe<Scalars['Boolean']>;
};

export type Photo = {
  __typename?: 'Photo';
  id: Scalars['String'];
  thumbURL?: Maybe<Scalars['String']>;
  photoURL?: Maybe<Scalars['String']>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type Query = {
  __typename?: 'Query';
  /**
   * If filter is true, it will filter user with email, nickname or name.
   * You can add pagination with first and after args.
   */
  users?: Maybe<UsersConnection>;
  user?: Maybe<User>;
  me?: Maybe<User>;
  messages: Array<Message>;
  channels: Array<Channel>;
  friends: Array<User>;
  galleries: Array<Gallery>;
};


export type QueryUsersArgs = {
  user?: Maybe<UserQueryInput>;
  includeUser?: Maybe<Scalars['Boolean']>;
  filter?: Maybe<Scalars['Boolean']>;
  first?: Maybe<Scalars['Int']>;
  last?: Maybe<Scalars['Int']>;
  before?: Maybe<Scalars['String']>;
  after?: Maybe<Scalars['String']>;
};


export type QueryUserArgs = {
  id: Scalars['ID'];
};


export type QueryGalleriesArgs = {
  userId: Scalars['String'];
};

export type Reaction = {
  __typename?: 'Reaction';
  id: Scalars['ID'];
  type?: Maybe<Scalars['String']>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type Reply = {
  __typename?: 'Reply';
  id: Scalars['String'];
  sender?: Maybe<User>;
  type?: Maybe<Scalars['String']>;
  text?: Maybe<Scalars['String']>;
  filePath?: Maybe<Scalars['String']>;
  replies?: Maybe<Array<Maybe<Reply>>>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type SocialUserInput = {
  socialId: Scalars['String'];
  authType: AuthType;
  email?: Maybe<Scalars['String']>;
  photoURL?: Maybe<Scalars['String']>;
  thumbURL?: Maybe<Scalars['String']>;
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  phone?: Maybe<Scalars['String']>;
};

export type Subscription = {
  __typename?: 'Subscription';
  userSignedIn?: Maybe<User>;
  userUpdated?: Maybe<User>;
  friendChanged?: Maybe<FriendSub>;
};


export type SubscriptionUserSignedInArgs = {
  userId: Scalars['ID'];
};


export type SubscriptionUserUpdatedArgs = {
  userId: Scalars['ID'];
};


export type SubscriptionFriendChangedArgs = {
  userId: Scalars['ID'];
};


export type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  email?: Maybe<Scalars['String']>;
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  thumbURL?: Maybe<Scalars['String']>;
  photoURL?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  socialId?: Maybe<Scalars['String']>;
  authType?: Maybe<AuthType>;
  phone?: Maybe<Scalars['String']>;
  verified?: Maybe<Scalars['Boolean']>;
  statusMessage?: Maybe<Scalars['String']>;
  isOnline?: Maybe<Scalars['Boolean']>;
  lastSignedIn?: Maybe<Scalars['DateTime']>;
  notifications?: Maybe<Array<Maybe<Notification>>>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type UserEdge = {
  __typename?: 'UserEdge';
  node?: Maybe<User>;
  cursor?: Maybe<Scalars['String']>;
};

export type UserInput = {
  email: Scalars['String'];
  password: Scalars['String'];
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  phone?: Maybe<Scalars['String']>;
  statusMessage?: Maybe<Scalars['String']>;
};

export enum UserModeType {
  Default = 'DEFAULT',
  Hidden = 'HIDDEN',
  Block = 'BLOCK'
}

export type UserProfileInput = {
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  phone?: Maybe<Scalars['String']>;
  thumbURL?: Maybe<Scalars['String']>;
  photoURL?: Maybe<Scalars['String']>;
  statusMessage?: Maybe<Scalars['String']>;
};

export type UserQueryInput = {
  email?: Maybe<Scalars['String']>;
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  phone?: Maybe<Scalars['String']>;
};

/**
 * Simple wrapper around our list of launches that contains a cursor to the
 * last item in the list. Pass this cursor to the launches query to fetch results
 * after these.
 */
export type UsersConnection = {
  __typename?: 'UsersConnection';
  totalCount?: Maybe<Scalars['Int']>;
  edges?: Maybe<Array<Maybe<UserEdge>>>;
  pageInfo?: Maybe<PageInfo>;
};
