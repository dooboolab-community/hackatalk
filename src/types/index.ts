import { StyleProp, TextStyle } from 'react-native';

import { DefaultTheme } from 'styled-components';
import { SFC } from 'react';

export interface User {
  uid: string;
  displayName: string;
  photoURL: string;
  statusMsg: string;
}

interface IconProps {
  style?: StyleProp<TextStyle>;
  width?: number | string;
  height?: number | string;
  children?: never;
}

export type IconType = SFC<IconProps>;

export enum ThemeType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}
