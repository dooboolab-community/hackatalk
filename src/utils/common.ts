export const validateEmail = (email: string): boolean => {
  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(email);
};

export const validatePassword = (password: string): boolean => {
  const re = /^.*(?=.{6,15})(?=.*[0-9])(?=.*[a-zA-Z])(?=.*[!@#$%^&+=]).*$/;
  return re.test(password);
};
