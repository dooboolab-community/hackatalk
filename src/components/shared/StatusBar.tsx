import { AppContext } from '../../providers/AppProvider';
import React from 'react';
import { StatusBar } from 'react-native';
import { ThemeType } from '../../types';

// prettier-ignore
function Shared() {
  const { state } = React.useContext(AppContext);
  const statusColor = state.theme === ThemeType.LIGHT ? 'dark-content' : 'light-content';

  return <StatusBar barStyle={statusColor} />;
}

export default Shared;
