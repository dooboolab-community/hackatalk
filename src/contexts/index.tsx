import {
  ProfileModalAction,
  ProfileModalState,
  initialProfileModal,
  profileModalReducer,
} from '../reducers/ProfileModalReducer';
import React, {
  Dispatch,
  FC,
  ReactNode,
  createContext,
  useContext,
  useReducer,
  useRef,
} from 'react';
import {
  ThemeAction,
  initialTheme,
  themeReducer,
} from '../reducers/themeReducer';

import { ThemeType } from '../types';

// TODO: preferred to invent a way to automatically overload or joined typescript perk
interface MainState {
  theme: ThemeType;
  profileModal: ProfileModalState;
}

type TMainAction = ThemeAction | ProfileModalAction;

const mainInitialState = {
  theme: initialTheme,
  profileModal: initialProfileModal,
};

const mainReducer = (
  { theme, profileModal }: MainState,
  action: TMainAction,
): MainState => {
  // middleware goes here, i.e. calling analytics service, etc.
  return {
    theme: themeReducer(theme, action),
    profileModal: profileModalReducer(profileModal, action),
  };
};

export const StateContext = createContext<[MainState, Dispatch<TMainAction>]>([
  mainInitialState,
  (): null => null,
]);

export const StateProvider: FC<ReactNode> = ({ children }) => {
  const modal = useRef<any>(null);

  const modalInit = (): MainState => {
    return {
      ...mainInitialState,
      profileModal: {
        ...mainInitialState.profileModal,
        modal,
      },
    };
  };
  return (
    <StateContext.Provider
      value={useReducer(mainReducer, mainInitialState, modalInit)}
    >
      {children}
    </StateContext.Provider>
  );
};

export const useStateValue = (): [MainState, Dispatch<TMainAction>] =>
  useContext(StateContext);
