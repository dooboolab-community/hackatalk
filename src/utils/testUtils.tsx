import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
