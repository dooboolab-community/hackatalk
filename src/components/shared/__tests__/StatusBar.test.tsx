import 'react-native';

import * as React from 'react';

import { createTestElement, createTestProps } from '../../../utils/testUtils';

import { StateProvider } from '../../../contexts';
import StatusBar from '../StatusBar';
import { ThemeProvider } from 'styled-components/native';
import { ThemeType } from '../../../types';
import { createTheme } from '../../../theme';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let props: any;
let component: React.ReactElement;
// let testingLib: RenderResult;

describe('[StatusBar] render', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<StatusBar {...props} />);
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON | null = renderer
      .create(component)
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });

  // describe('interactions', () => {
  //   beforeEach(() => {
  //     testingLib = render(component);
  //   });

  //   it('should simulate onPress', () => {
  //     const btn = testingLib.queryByTestId('btn');
  //     act(() => {
  //       fireEvent.press(btn);
  //     });
  //     expect(cnt).toBe(3);
  //   });
  // });
});
