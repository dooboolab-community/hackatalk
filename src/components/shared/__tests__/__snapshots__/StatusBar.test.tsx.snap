// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[StatusBar] render renders without crashing 1`] = `null`;
